package/kedro_viz/api/rest/responses/run_events.py

```diff
@@ -0,0 +1,233 @@
+"""Response for run events API endpoint."""
+
+import json
+import logging
+import re
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Optional, Any, Union, Set, Tuple
+
+from pydantic import BaseModel, Field
+
+from kedro_viz.constants import VIZ_METADATA_ARGS
+from kedro_viz.launchers.utils import _find_kedro_project
+
+logger = logging.getLogger(__name__)
+
+class NodeInfo(BaseModel):
```

Comment:
I was thinking maybe we change NodeStatus and DatasetStatus to ENUM 
NodeStatus could be 'Success', 'Fail', 'Skipped' 
DatasetStatus could be 'Available', and 'Missing' for now.

Reply:
as @Huongg mentioned; let's align on correct states first and then make this change